src/ivc/fold_relaxed_plonk_instance_chip.rs

```diff
@@ -1155,6 +1179,169 @@ mod tests {
         }
     }
 
+    #[test_log::test]
+    fn fold_instances_test() {
+        let Fixture {
+            mut td,
+            config,
+            mut rnd,
+            r,
+            ..
+        } = Fixture::default();
+
+        let mut layouter = SingleChipLayouter::new(&mut td, vec![]).unwrap();
+
+        let mut relaxed_plonk = RelaxedPlonkInstance::<C1>::new(2, 0, 0);
+
+        let bn_chip = BigUintMulModChip::<Base>::new(
+            config
+                .into_smaller_size::<{ big_uint_mul_mod_chip::MAIN_GATE_T }>()
+                .unwrap(),
+            LIMB_WIDTH,
+            LIMB_LIMIT,
+        );
+
+        for _round in 0..=NUM_OF_FOLD_ROUNDS {
+            let input_instances = [ScalarExt::random(&mut rnd), ScalarExt::random(&mut rnd)];
+
+            let on_circuit_instances_cell = layouter.assign_region(
+                || "fold instances test",
+                |region| {
+                    let mut ctx = RegionCtx::new(region, 0);
+
+                    let mut advice_columns_assigner = config.advice_cycle_assigner();
+
+                    macro_rules! assign_scalar_as_bn {
```

Comment:
Does it worth a separate utility function? 
I saw it used in `fold_challenge_test` as well. If just two places,  we don't need to separate.

Reply:
I was going to summarize everything overused at the end, once all the tests are done (including the overall `fold`, since there are a lot of the same constructs here). What can be allocated to a function is allocated to a function, what's in macro is allocated to a macro. 

There's a lot of copy-paste here so far, but it's meaningful to see how large chunks of code can be generalized